### ISSUE TITLE
Markdown newlines are different from Element

### ISSUE THREAD

**Issue by notramo** [ajbura/cinny#193]
The original Markdown spec describes that newlines should be removed from the input, unless 2 spaces are appended at the end (line break), or more than 1 newlines (new paragraph).
However, many applications use a different flavor, e.g. Github, or Element, where every newline is persisted as is.
This is useful in instant messaging.
Cinny should also preserve these line breaks.
![image](https://user-images.githubusercontent.com/26059643/146675680-77754fe9-d6a9-4802-92e5-8a7c72ba819a.png)
![image](https://user-images.githubusercontent.com/26059643/146675707-e78169bb-4824-47a8-99a5-0bb280a4d80f.png)

**Comment by samueldr** [comment]
This is actually element being buggy.

See the discussion here:

 - https://github.com/ajbura/cinny/issues/174#issuecomment-984271524

TLDR: when there is no markdown in the input, element sends the input as plain text. If you use, e.g. `*hi*` in one of those lines, to force markdown, element will behave exactly as Cinny does.

**Comment by ShadowJonathan** [comment]
> The original Markdown spec describes that newlines should be removed from the input

Could you give me a link to that?

**Comment by samueldr** [comment]
AFAICT it's not part of the *original* markdown spec *as described in the first post of this issue*.

 - https://daringfireball.net/projects/markdown/basics

It states that “*A paragraph is simply one or more consecutive lines of text, separated by one or more blank lines*”, nothing about two spaces to make a newline, which is a common extension from the *original* spec.

But I feel I need to remind everyone here that *cinny is doing things right*. It is element that has a buggy behaviour and sometimes sends plaintext messages. When there is markdown formatting *other than linebreaks*, element will behave differently.
